hash_practice/exercises.py

```diff
@@ -2,18 +2,36 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(1)
     """
-    pass
+    anagram_dict = {}
+
+    for string in strings: 
+        word_sorted = "".join(sorted(string))
+        if word_sorted not in anagram_dict: 
+            anagram_dict[word_sorted] = [string]
+        else: 
+            anagram_dict[word_sorted] = [string]
+
+    return list(anagram_dict.values())
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(1)

```

Comment:
Likewise, in this case we add an entry to `most_common` for each number in `nums` so the space complexity here is O(n).